docs/_tutorials/reference.md

```diff
@@ -0,0 +1,141 @@
+| `client` | All listeners | Web API client that uses the token associated with that event. For single-workspace installations, the token is provided to the constructor. For multi-workspace installations, the token is returned by the `authorize` function.
+| `respond` | `shortcut`, `action`, `command` | Function that responds to an incoming event **if** it contains a `response_url`. `respond` returns a promise that resolves with the results of responding using the `response_url`. For shortcuts, `respond` will **only** work for message shortcuts (not global shortcuts).
+| `context` | All listeners | Event context. This object contains data about the event and the app, such as the `botId`. Middleware can add additional context before the event is passed to listeners.
+| `body` | All listeners | Object that contains the entire body of the request (superset of `payload`). Some accessory data is only available outside of the payload (such as `trigger_id` and `authorizations`).
+
+#### Body and payload references
+The structure of the `payload` and `body` is detailed on the API site:
+- `action`: [`body`](https://api.slack.com/reference/interaction-payloads/block-actions) and [`payload`s](https://api.slack.com/reference/block-kit/block-elements)
+- `event`: [`body`](https://api.slack.com/types/event) and [`payload`s](https://api.slack.com/events)
+- `shortcut`: [`body` and `payload`](https://api.slack.com/reference/interaction-payloads/shortcuts)
+- `command`: [`body` ](https://api.slack.com/interactivity/slash-commands)
+- `view`: [`view_submission` `body` and `payload`](https://api.slack.com/reference/interaction-payloads/views#view_submission); [`view_closed` `body` and `payload`](https://api.slack.com/reference/interaction-payloads/views#view_closed)
+- `options`: [`body` and `payload`](https://api.slack.com/reference/block-kit/block-elements#external_select)
+
+### Difference from listener middleware
+Listener middleware is used to implement logic across many listener functions (though usually not all of them). Listener middleware has the same arguments as the above listener functions, with one distinction: they also have a `next()` function that **must** be called in order to pass the chain of execution. Learn more about listener middleware [in the documentation](/bolt-js/concepts#listener-middleware).
+
+## Initialization options
+Bolt includes a collection of initialization options to customize apps. There are two primary kinds of options: Bolt app options and receiver options. The receiver options may change based on the receiver your app uses. The following receiver options are for the default `ExpressReceiver` (so they'll work as long as you aren't using a custom receiver).
+
+### Receiver options
+`ExpressReceiver` options can be passed into the app constructor, just like the Bolt app options. They'll be passed to the `ExpressReceiver` instance upon initialization.
+
+| Option  | Description  |
```

Comment:
It would be nice to know which are required or optional options. I noticed that some descriptions start with "Optional,...". Would it be better to just have a column "Required"?